x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/analytics/AnalyticsCollectionService.java

```diff
@@ -0,0 +1,128 @@
+        Client client,
+        ClusterService clusterService,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        this.clientWithOrigin = new OriginSettingClient(client, ENT_SEARCH_ORIGIN);
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        clusterService.addListener(this);
+    }
+
+    /**
+     * Retrieve an analytics collection by name {@link AnalyticsCollection}
+     *
+     * @param collectionName {@link AnalyticsCollection} name.
+     * @param listener The action listener to invoke on response/failure.
+     */
+    public void getAnalyticsCollection(String collectionName, ActionListener<AnalyticsCollection> listener) {
+        if (analyticsCollections.containsKey(collectionName) == false) {
+            listener.onFailure(new ResourceNotFoundException(collectionName));
+            return;
+        }
+
+        listener.onResponse(analyticsCollections.get(collectionName));
+    }
+
+    /**
+     * Create a new {@link AnalyticsCollection}
+     *
+     * @param analyticsCollection {@link AnalyticsCollection} to be created.
+     * @param listener The action listener to invoke on response/failure.
+     */
+    public void createAnalyticsCollection(AnalyticsCollection analyticsCollection, ActionListener<AnalyticsCollection> listener) {
+        if (analyticsCollections.containsKey(analyticsCollection.getName())) {
+            listener.onFailure(new ResourceAlreadyExistsException(analyticsCollection.getName()));
+            return;
+        }
+
+        CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(
+            analyticsCollection.getEventDataStream()
+        );
+
+        ActionListener<AcknowledgedResponse> createDataStreamListener = ActionListener.wrap(
+            resp -> listener.onResponse(analyticsCollection),
+            listener::onFailure
+        );
+
+        clientWithOrigin.execute(CreateDataStreamAction.INSTANCE, createDataStreamRequest, createDataStreamListener);
+    }
+
+    /**
+     * Delete an analytics collection by name {@link AnalyticsCollection}
+     *
+     * @param collectionName {@link AnalyticsCollection} name.
+     * @param listener The action listener to invoke on response/failure.
+     */
+    public void deleteAnalyticsCollection(String collectionName, ActionListener<AcknowledgedResponse> listener) {
+        if (analyticsCollections.containsKey(collectionName) == false) {
+            listener.onFailure(new ResourceNotFoundException(collectionName));
+            return;
+        }
+
+        DeleteDataStreamAction.Request deleteDataStreamRequest = new DeleteDataStreamAction.Request(
+            analyticsCollections.get(collectionName).getEventDataStream()
+        );
+
+        clientWithOrigin.execute(DeleteDataStreamAction.INSTANCE, deleteDataStreamRequest, listener);
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
```

Comment:
Can we check that the metadata changed to skip the work if it's not needed (nothing changed)? You can check with `event.metadataChanged`.

Reply:
Added an early return if `event. metadataChanged()` returns false.